tests/test_cli_replace_modal_subgraph.py

```diff
@@ -0,0 +1,195 @@
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    expected_new_link_id = sub_network["expected_bike_link_id"]
+    assert output_network.has_link(
+        expected_new_link_id
+    ), f"Link {expected_new_link_id} is missing from the output network"
+    assert (
+        output_network.link(expected_new_link_id)["capacity"] == 9999
+    ), f"Link {expected_new_link_id} did not have capacity increased"
+
+    link_id = network["link_id"]
+    assert output_network.has_link(link_id), f"Link {link_id} is missing from the output network"
+    assert (
+        output_network.link(link_id)["capacity"] == original_net_capacity[link_id]
+    ), f"Link {link_id} did not retain th original capacity value"
+
+
+def test_does_not_increase_capacity_if_not_requested(tmpdir, network, sub_network):
+    original_subnet_capacity = {
+        link_id: data["capacity"] for link_id, data in sub_network["network"].links()
+    }
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=False,
+    )
+
+    for link_id, old_capacity in original_subnet_capacity.items():
+        expected_new_link_id = sub_network["expected_link_id_mapping"][link_id]
+        assert output_network.has_link(
+            expected_new_link_id
+        ), f"Link {expected_new_link_id} is missing from the output network"
+        assert (
+            output_network.link(expected_new_link_id)["capacity"] == old_capacity
+        ), f"Link {expected_new_link_id} did not retain th original capacity value"
+
+
+def test_adds_nodes_only_once_with_multiple_subgraphs(tmpdir, network, sub_network):
+    original_net_no_nodes = len(list(network["network"].nodes()))
+    original_subnet_no_nodes = len(list(sub_network["network"].nodes()))
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike,bus",
+        increase_capacity=False,
+    )
+
+    assert (
+        len(list(output_network.nodes())) == original_net_no_nodes + original_subnet_no_nodes
+    ), "Number of nodes in the output network is not consistent with the input networks"
+
+
+def test_adds_links_shared_in_subgraph_network_separately(tmpdir, network, sub_network):
+    original_net_no_links = len(list(network["network"].links()))
+    original_subnet_no_links = len(list(sub_network["network"].links()))
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike,bus",
+        increase_capacity=False,
+    )
+
+    assert len(list(output_network.links())) == original_net_no_links + (
```

Comment:
I can't understand from this test why we expect twice the number of links we see in the sub-network. Is there a way to make that clearer?